Objects/longobject.c

```diff
@@ -332,45 +332,48 @@ _PyLong_Negate(PyLongObject **x_p)
     Py_DECREF(x);
 }
 
+#define PYLONG_FROM_INT(UINT_TYPE, INT_TYPE, ival) \
+    do { \
+        PyLongObject *v; \
+        UINT_TYPE abs_ival, t; \
+        int ndigits; \
+        /* Handle small and medium cases. */ \
+        if (IS_SMALL_INT(ival)) { \
+            return get_small_int((sdigit)ival); \
+        } \
+        if (-(INT_TYPE)PyLong_MASK <= ival && ival <= (INT_TYPE)PyLong_MASK) { \
+            return _PyLong_FromMedium((sdigit)ival); \
+        } \
+        /* Count digits (at least two - smaller cases were handled above). */ \
+        abs_ival = ival < 0 ? 0U-(UINT_TYPE)ival : (UINT_TYPE)ival; \
+        /* Do shift in two steps to avoid possible undefined behavior. */ \
+        t = abs_ival >> PyLong_SHIFT >> PyLong_SHIFT; \
+        ndigits = 2; \
```

Comment:
```suggestion
        UINT_TYPE abs_ival = ival < 0 ? 0U-(UINT_TYPE)ival : (UINT_TYPE)ival; \
        /* Do shift in two steps to avoid possible undefined behavior. */ \
        UINT_TYPE t = abs_ival >> PyLong_SHIFT >> PyLong_SHIFT; \
        int ndigits = 2; \
```